src/Common/NamePrompter.h

```diff
@@ -102,6 +103,16 @@ class IHints
         return prompter.getHints(name, getAllRegisteredNames());
     }
 
+    String getHintsString(const String & name) const
+    {
+        auto hints = getHints(name);
+
+        /// Note: we don't use toString because it will cause writeCString naming conflict in src/Dictionaries/MongoDBDictionarySource.cpp
+        for (auto & hint : hints)
+            hint = "'" + hint + "'";
+        return !hints.empty() ? ", may be you meant: [" + boost::algorithm::join(hints, ",") + "]" : "";
```

Comment:
The message seems really specific, especially because it's an addition (starts with ", ")
I would rather define a local function in each file which appends to the error message hints if there are any (for example in `AlterCommands.cpp`.

Reply:
@antonio2368 I use this message not only in `AlterCommands`, but also `ColumnsDescription` and `ProjectionsDescription`. 
So I put it in `IHints`

Reply:
I refactored changes in a separate PR, hope you don't mind #35852

Reply:
That's ok.   Do I need to close this PR?

Reply:
No need.